fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -810,6 +814,18 @@ public CompletableFuture<Boolean> markReadable() {
         return getIndexer().markIndexReadable(true);
     }
 
+    enum IndexingStampOperation {
+        QUERY, BLOCK, UNBLOCK,
```

Comment:
I think it would be better to have separate methods (`queryIndexStamps`, `blockIndexBuilds`, `unblockIndexBuilds`). Specifically for `blockIndexBuilds` and `unblockIndexBuilds`, the fact that it is backed by the indexing stamp could be internal, and doesn't need to be exposed (conceptually) to the user.

That being said, if you feel adamant that these should be an enum as you have it, then:
1. These enum values should have javadoc, specifically important is that `BLOCK` will abort any currently running index builds
2. The method should have a javadoc
3. The method should be very-like, something like `performIndexingStampOperation`
4. The method should document the id. Particularly that it should be unique and provide enough information to identify who and why it was blocked.

Reply:
How about: 
```
indexingSessionQuery() 
indexingSessionBlock(@Nullable id, @Nullable ttlSeconds) 
indexingSessionUnblock(@Nullable id)
```
?

Reply:
Converted to the functions above.

Reply:
Why `indexingSession` instead of `indexingStamp`? I feel like I prefer `queryIndexingStamps` better. I can kind of see `blockIndexingSession`, but `blockIndexBuilds` and `unblockIndexBuilds` seems to be more indicative of what these methods do from a semantic sense (whereas something like `indexingStampBlock` is clearer on the mechanics (setting a flag in the indexing stamp), but it doesn't necessarily inform the user why they might want to call this method).

I mostly feel strong that the verbs should come first in those names, especially `block` and `unblock`, as that highlights more clearly that these are operations that have side-effects

Reply:
I changed to your suggestion, but wouldn't `blockIndexingContinuation` be more accurate?

Reply:
The way I see it, blocking prevents continuing an index build (for future and ongoing indexing processes). 
`.setAllowUnblock(true)` does not ignore the block but unblocks and continues indexing in one "atomic" transaction. The motivation was to prevent other indexing processes (possibly with undesired arguments) from taking over the indexing task between the `unblock` and `buildIndex` user calls. 
But it's not a big deal. I can live with both names.

Reply:
Ah, I see. I think there's maybe something missing, then, because I could see a scenario where we, for example, want to run an index build (maybe using one specific set of parameters), but we don't want other builds to try and pre-empt it. In that case, I'd expect us to set the `block` on the index, and then configure the one-specific index builder to ignore the block, but to leave it in place. Is that not supported with the current API, then?

Reply:
No, it is not. But we can always use `setUseSynchronizedSession` to ensure a single indexing process. 
(But not a handful of mutual ones...)